tests/test_config_flow.py

```
@@ -231,25 +202,71 @@ async def _raiser(*args, **kwargs) -> Never:
+    assert err.value is exc
 
-def test_log_and_set_error_sets_base(caplog: pytest.LogCaptureFixture) -> None:
-    """_log_and_set_error should log the message and set errors['base']."""
+
+@pytest.mark.asyncio
+async def test_validate_input_timeout_uses_connect_timeout_error(
+    monkeypatch: pytest.MonkeyPatch, caplog: pytest.LogCaptureFixture
+) -> None:
+    """OPNsense timeout should map to connect_timeout and redact credentials in logs."""
+    username = "admin"
+    password = "supersecret"
```

Ruff (0.15.21)

[error] Possible hardcoded password assigned to: "password"

(S105)